docs/changes/modeling/16673.api.rst

```diff
@@ -0,0 +1,2 @@
+Subclasses of ``_NonLinearLSQFitter``, so any subclasses of the public ``LevMarLSQFitter``, ``TRFLSQFitter``, ``LMLSQFitter`` or ``DogBoxLSQFitter``, should now accept an additional ``fit_param_indices`` kwarg in their function signatures.
+Nothing is needed to be done with this kwarg, and it might not be set, but it can optionally be passed through to ``fitter_to_model_params_array`` for a performance improvement.
```

Comment:
Perhaps it would make sense to say here somewhere that in fact they should now be able to accept additional keyword arguments, to make sure that people don't just add ``fit_param_indices=`` but also e.g. ``**kwargs``. Basically write it in a way that it will be future proof if we need to add more? (Or we could revert to doing the single dictionary as we had it before)

Reply:
Is this resolved yet?

Reply:
This makes sense to me.

Reply:
Fixed